Session08/binarytree/solutions/BinarySearchTree.java

```diff
@@ -0,0 +1,123 @@
+// This class fails to compile, can you figure out why, and fix it?
+// This is beacuse BinaryTree has a constructor which is not default. Hence, when we extend this class and try to call a no-argument constructor, it'll give a compile-time error. so, we call it by passing the parameters (to instantiate the root node) in super() called inside the constructor of BinarySearchTree
+
+class BinarySearchTree extends BinaryTree {
+
+    private int flag = 0;
+    
+    public BinarySearchTree()
+    {
+        super(0);
+    }
+    
+    // insert node
+	@Override
+	public boolean insert(int value) {
+		super.root = insert(super.root, value);
+		return true;
+	}
+
+    private TreeNode insert(TreeNode root, int data)
+    {
+        // if current node is null, insert node with data
+        if (root == null) root = new TreeNode(data);
```

Comment:
If both your if and else blocks have a single line, this formatting is fine, like the way you have used below.

However, in this case, please adhere to the repo's coding style of

```java
if (<condition>) {
}
else {
}
```

Curly braces are good. Always use them, even when they are not needed.